### DISCUSSION TITLE
Can't use/activate modifiers and layers?

### DISCUSSION THREAD

**Discussion by jerdaw** [manna-harbour/miryoku#140] [Q&A] [answered]
[Using qmk's shell.nix on NixOS with a lily58]

I'm not sure if this is a bug or if I've just missed something, but I'm unable to actually activate any of the modifier or layers after going through the setup instructions and flashing. All the base layer keys (alpha, space, enter, period, etc.) work just fine. However, if I hold down any of the modifiers or layer buttons, it just repeats the base key that is at that location.

E.g., holding down space and trying to use the directional keys just types "    n  e     i    o  ". No directional key is actually entered.
E.g. 2, trying to type "T" by holding n for shift and then pressing t just types something like "nnnntn" (doesn't seem to make a difference how long I hold n before pressing "T").

I haven't modified miryoku_qmk in any way after git fetching and merging it per the readme.org

To compile and flash, I just used the basic qmk commands:
```
qmk compile -c -kb lily58/rev1 -km manna-harbour_miryoku
qmk flash -c -kb lily58/rev1 -km manna-harbour_miryoku
```

Any help is appreciated.

**Comment by manna-harbour** [comment]
I just wrote a comprehensive answer but my phone ate it and I don't have time now to re-write it. 😅 The short answer is do an EEPROM erase in QMK Toolbox.

**Reply by jerdaw** [reply]
Thanks for the reply! I'm still working on getting qmk toolbox to recognize the HID console device but, once I figure that out, I'll make sure to report on how it goes.

**Reply by jerdaw** [reply]
Managed to get it to work in Linux, and clearing EEPROM solved it immediately.  Thanks for the help!

**Comment by manna-harbour** [comment]
On Linux you can do `avrdude -p m32U4 -P /dev/ttyACM0 -c avr109 -e` to erase a pro-micro. Otherwise, flash your previous firmware and set the persistent default layer to layer 0.

**Comment by manna-harbour** [comment, marked as answer]
That's normal behaviour for the Tap layer. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features. Tap requires a reset or power cycle on master to exit. You can verify you're on the Tap layer by changing the alpha arrangement. E.g. add `-e MIRYOKU_TAP=QWERTY` to the `qmk` command line. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#tap.

The keyboard should always start on Base. If it starts on another layer it's because there is a persistent default layer set from the previous firmware. To fix it you can erase the EEPROM with QMK Toolbox on Windows or Mac, or with `avrdude -p m32U4 -P /dev/ttyACM0 -c avr109 -e` for a pro-micro on Linux. Alternatively, flash the previous firmware and set the persistent default layer to layer 0 using whatever method is set up in that firmware. See https://github.com/qmk/qmk_firmware/blob/master/docs/ref_functions.md#setting-the-persistent-default-layer, https://github.com/qmk/qmk_firmware/blob/master/docs/ref_functions.md#wiping-the-eeprom-persistent-storage, https://docs.keeb.io/reset-eeprom.
